web/source/text/prediction/modelManager.ts

```diff
@@ -59,6 +59,12 @@ namespace com.keyman.text.prediction {
       // Register the model for each targeted language code variant.
       let mm = this;
       model.languages.forEach(function(code: string) {
+        // Prevent null / undefined codes; they're invalid.
+        if(!code) {
+          console.warn("Null / undefined language codes are not permitted for registration.");
```

Comment:
```suggestion
          let codes; try { codes = JSON.stringify(model.languages); } catch(e) { codes = e; }
          console.warn(`Null / undefined language codes are not permitted for registration (was passed: ${codes}).`);
```

Yeah yeah I know I am abusing the try catch except message but it's just safer than blindly trying to json stringify an object and this way if this ever happened we could figure out what was going wrong even if the message is weird!

Better would be a wrappered function:

```ts
function debugJson(o: any) {
  try {
    return JSON.stringify(o);
  } catch(e) {
    return JSON.stringify({debugJsonError: 
       e instanceof Error ? e.message :
       typeof e == 'object' ? 'object thrown' :
       e.toString()});
  }
}
```

There are probably more betterer functions online yet even. Just trying to make an incremental improvement to our debug tracking for the future here so feel free to dump this into a "wishful thinking future issue"!